src/ports/postgres/modules/convex/mlp.sql_in

```diff
@@ -0,0 +1,744 @@
+  also contain id_col_name used for identifying each row.</DD>
+
+  <DT>id_col_name</DT>
+  <DD>TEXT. The name of the id column in the input table.</DD>
+
+  <DT>output_table</DT>
+  <DD>TEXT. Name of the table where output predictions are written. If this
+table name is already in use, then an error is returned.  Table contains:</DD>
+    <table class="output">
+      <tr>
+        <th>id</th>
+        <td>Gives the 'id' for each prediction, corresponding to each row from the data_table.</td>
+      </tr>
+      <tr>
+        <th>estimated_<COL_NAME></th>
+        <td>
+        (For pred_type='response') The estimated class
+         for classification or value for regression, where
+         <COL_NAME> is the name of the column to be
+         predicted from training data
+        </td>
+      </tr>
+      <tr>
+        <th>prob_<CLASS></th>
+        <td>
+        (For pred_type='prob' for classification) The
+        probability of a given class <CLASS> as given by
+        softmax. There will be one column for each class
+        in the training data.
+        </td>
+      </tr>
+
+
+  <DT>pred_type</DT>
+  <DD>TEXT.
+
+the type of output requested:
+'response' gives the actual prediction,
+'prob' gives the probability of each class.
+for regression, only type='response' is defined.
+The name of the id column in the input table.</DD>
+</DL>
+</table>
+
+@anchor example
+@par Examples
+-#  Create an input data set.
+<pre class="example">
+CREATE TABLE iris_data(
+    id integer,
+    attributes numeric[],
+    class_text varchar,
+    class integer
+);
+INSERT INTO iris_data VALUES
+(1,ARRAY[5.1,3.5,1.4,0.2],'Iris-setosa',1),
+(2,ARRAY[4.9,3.0,1.4,0.2],'Iris-setosa',1),
+(3,ARRAY[4.7,3.2,1.3,0.2],'Iris-setosa',1),
+(4,ARRAY[4.6,3.1,1.5,0.2],'Iris-setosa',1),
+(5,ARRAY[5.0,3.6,1.4,0.2],'Iris-setosa',1),
+(6,ARRAY[5.4,3.9,1.7,0.4],'Iris-setosa',1),
+(7,ARRAY[4.6,3.4,1.4,0.3],'Iris-setosa',1),
+(8,ARRAY[5.0,3.4,1.5,0.2],'Iris-setosa',1),
+(9,ARRAY[4.4,2.9,1.4,0.2],'Iris-setosa',1),
+(10,ARRAY[4.9,3.1,1.5,0.1],'Iris-setosa',1),
+(11,ARRAY[7.0,3.2,4.7,1.4],'Iris-versicolor',2),
+(12,ARRAY[6.4,3.2,4.5,1.5],'Iris-versicolor',2),
+(13,ARRAY[6.9,3.1,4.9,1.5],'Iris-versicolor',2),
+(14,ARRAY[5.5,2.3,4.0,1.3],'Iris-versicolor',2),
+(15,ARRAY[6.5,2.8,4.6,1.5],'Iris-versicolor',2),
+(16,ARRAY[5.7,2.8,4.5,1.3],'Iris-versicolor',2),
+(17,ARRAY[6.3,3.3,4.7,1.6],'Iris-versicolor',2),
+(18,ARRAY[4.9,2.4,3.3,1.0],'Iris-versicolor',2),
+(19,ARRAY[6.6,2.9,4.6,1.3],'Iris-versicolor',2),
+(20,ARRAY[5.2,2.7,3.9,1.4],'Iris-versicolor',2);
+</pre>
+-# Generate a multilayer perceptron with a single hidden layer of 5 units.
+Use the attributes column as the independent variables, and use the class
+column as the classification. Set the tolerance to 0 so that 5000
+iterations will be run. Use a hyperbolic tangent activation function.
+The model will be written to mlp_model.
+<pre class="example">
+SELECT madlib.mlp_classification(
```

Comment:
In case user attempts to run multiple times, precede `SELECT` statement with:

```
DROP TABLE IF EXISTS mlp_model;
DROP TABLE IF EXISTS mlp_model_summary;
```